src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/_params.py

```diff
@@ -8,28 +8,72 @@
+
+register_cli_argument('batch pool reset', 'json_file', type=file_type, help='The file containing PoolUpdatePropertiesParameter object in JSON format, if this parameter is specified, all other parameters are ignored.', validator=validate_json_file, completer=FilesCompleter())
+register_cli_argument('batch pool reset', 'pool_id', help='The ID of the pool to be updated.')
+register_cli_argument('batch pool reset', 'application_package_references', nargs='+', type=application_package_reference_format)
+register_cli_argument('batch pool reset', 'certificate_references', nargs='+', type=certificate_reference_format)
+register_cli_argument('batch pool reset', 'metadata', nargs='+', type=metadata_item_format)
+
+register_cli_argument('batch job list', 'filter', help=' An OData $filter clause.', arg_group='OData')
+register_cli_argument('batch job list', 'select', help=' An OData $select clause.', arg_group='OData')
+register_cli_argument('batch job list', 'expand', help=' An OData $expand clause.', arg_group='OData')
+register_cli_argument('batch job list', 'job_schedule_id', help='The ID of the job schedule from which you want to get a list of jobs.')
+
+register_cli_argument('batch certificate', 'thumbprint', help='The certificate thumbprint.')
+register_cli_argument('batch certificate', 'thumbprint_algorithm', help='The certificate thumbprint algorithm.')
+register_cli_argument('batch certificate', 'password', help='The password to access the certificate\'s private key.')
+register_cli_argument('batch certificate', 'cert_file', type=file_type, help='The certificate file: cer file or pfx file.', validator=validate_cert_file, completer=FilesCompleter())
+register_cli_argument('batch certificate delete', 'abort', action='store_true', help='Cancel the failed certificate deletion operation.')
+
+register_cli_argument('batch task create', 'json_file', type=file_type, help='The file containing the task(s) to create in JSON format, if this parameter is specified, all other parameters are ignored.', validator=validate_json_file, completer=FilesCompleter())
+register_cli_argument('batch task create', 'application_package_references', nargs='+', help='The space separated list of ids specifying the application packages to be installed.', type=application_package_reference_format)
+register_cli_argument('batch task create', 'job_id', help='The ID of the job containing the task.')
+register_cli_argument('batch task create', 'task_id', help='The ID of the task.')
+
+for item in ['batch certificate delete', 'batch certificate create', 'batch pool resize', 'batch pool reset', 'batch job list', 'batch task create']:
+    register_extra_cli_argument(item, 'account_name', arg_group='Batch Account',
+                                validator=validate_client_parameters,
+                                help='Batch account name. Environment variable: AZURE_BATCH_ACCOUNT')
```

Comment:
Consider using `az_config`, the CLI configuration feature for this.

This would allow the batch account to be set with `AZURE_BATCH_ACCOUNT` (as required) but also allow setting the config in the CLI configuration file.

For example https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py#L23.

Another example https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py#L51.